test/test_producer.py

```diff
@@ -81,11 +86,16 @@ def test_kafka_producer_gc_cleanup():
 
 
 @pytest.mark.skipif(not env_kafka_version(), reason="No KAFKA_VERSION set")
-@pytest.mark.parametrize("compression", [None, 'gzip', 'snappy', 'lz4'])
+@pytest.mark.parametrize("compression", [None, 'gzip', 'snappy', 'lz4', 'zstd'])
 def test_kafka_producer_proper_record_metadata(kafka_broker, compression):
+    if compression == 'zstd' and env_kafka_version() < (2, 1, 0):
+        return
```

Comment:
see note above about using a decorator instead to actually skip the test rather than considering it as passed

Reply:
Cannot figure a way to make this work without the conditional , since there is no (apparent to me) way of accessing the parameter injected by one decorator into the other one.